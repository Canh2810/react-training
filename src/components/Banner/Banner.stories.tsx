import { StoryObj, Meta } from '@storybook/react'
import Banner from '.'
import { POST_MOCK } from '@constants'

export default {
  title: 'Components/Banner',
  component: Banner,
} as Meta

type Story = StoryObj<typeof Banner>

export const Default: Story = {
  args: {
    post: POST_MOCK,
  },
}
