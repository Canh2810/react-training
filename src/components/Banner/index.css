.banner {
  position: relative;
  margin: 24px 0 114px 0;
}

.banner__img {
  width: 100%;
  height: auto;
  border-radius: 12px;
}

.banner__card {
  position: absolute;
  left: 64px;
  bottom: -68px;
  background-color: var(--background-primary-color);
  border-radius: 12px;
  border: 1px solid var(--border-color);
  padding: 40px;
  width: 589px;
  display: flex;
  flex-direction: column;
  align-items: start;
  gap: 16px;
  box-shadow: 0px 12px 24px -6px rgba(24, 26, 42, 0.12);
}

.banner__card-footer {
  margin-top: 4px;
  display: flex;
  align-items: center;
  gap: 20px;
  margin-top: 8px;
}

.banner__author-avatar {
  width: 36px;
  height: 36px;
  border-radius: 50%;
  margin-right: -8px;
}
