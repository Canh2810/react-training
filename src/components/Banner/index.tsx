// Libs
import { memo } from 'react'

// Types
import { IPost, TypoVariants } from '@types'

// Components
import { Badge, Typography } from '../index'

// Styles
import './index.css'

export interface PostCardProps {
  post: IPost
}

const Banner = ({ post }: PostCardProps) => {
  const { title, imageURL, author, category, date } = post || {}
  const { username, avatar } = author || {}

  return (
    <section className="banner">
      <img className="banner__img" src={imageURL} alt={title} />
      <div className="banner__card">
        <Badge title={category} />
        <Typography variant={TypoVariants.HeadingExtraLarge}>
          {title}
        </Typography>
        <div className="banner__card-footer">
          <img className="banner__author-avatar" src={avatar} alt={username} />
          <Typography>{username}</Typography>
          <Typography>{date}</Typography>
        </div>
      </div>
    </section>
  )
}

export default memo(Banner)
