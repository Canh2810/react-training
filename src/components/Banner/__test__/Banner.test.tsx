import { render } from '@testing-library/react'
import Banner from '..'
import { POST_MOCK } from '@constants'

describe('Banner component', () => {
  it('should render correctly', () => {
    const { container } = render(<Banner post={POST_MOCK} />)
    expect(container).toMatchSnapshot()
  })
})
