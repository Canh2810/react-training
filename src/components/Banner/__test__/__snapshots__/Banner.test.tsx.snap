// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component should render correctly 1`] = `
<div>
  <section
    class="banner"
  >
    <img
      alt="The Impact of Technology on the Workplace: How Technology is Changing"
      class="banner__img"
      src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
    />
    <div
      class="banner__card"
    >
      <span
        class="badge"
        style="background-color: rgb(75, 107, 251); color: rgb(255, 255, 255);"
      >
        Technology
      </span>
      <p
        class=""
        style="font-family: work-sans; font-size: 36px; font-weight: 600; line-height: 40px; color: rgb(59, 60, 74);"
      >
        The Impact of Technology on the Workplace: How Technology is Changing
      </p>
      <div
        class="banner__card-footer"
      >
        <img
          alt="Tracey Wilson"
          class="banner__author-avatar"
          src="https://www.w3schools.com/w3images/avatar2.png"
        />
        <p
          class=""
          style="font-family: plus-jakarta-sans; font-size: 16px; font-weight: 500; line-height: 24px; color: rgb(59, 60, 74);"
        >
          Tracey Wilson
        </p>
        <p
          class=""
          style="font-family: plus-jakarta-sans; font-size: 16px; font-weight: 500; line-height: 24px; color: rgb(59, 60, 74);"
        >
          August 20, 2022
        </p>
      </div>
    </div>
  </section>
</div>
`;
